compiler/qsc_qasm3/src/semantic/ast.rs

```diff
@@ -1001,15 +1008,15 @@ impl Display for IncludeStmt {
 #[derive(Clone, Debug)]
 pub struct QubitDeclaration {
     pub span: Span,
-    pub qubit: Box<Ident>,
-    pub size: Option<Expr>,
+    pub symbol_id: SymbolId,
+    pub size: Option<(u32, Span)>,
```

Comment:
Does is make sense to have a separate QubitArrayDeclaration instead of combining the two?